### PR TITLE
Updates Persister example

### PR DESCRIPTION
**Pull request by ConorOkus**

### CODE REVIEW

File: docs/tutorials/build_a_node_in_rust.md [hunk 1]
```diff
@@ -693,28 +693,37 @@ fn handle_ldk_event(..) {
 
 **References:** [`Event` docs](https://docs.rs/lightning/*/lightning/util/events/enum.Event.html), [LDK sample node event handling example](https://github.com/lightningdevkit/ldk-sample/blob/bc07db6ca4a3323d8718a27f85182b8157a20750/src/main.rs#L101-L240), [`EventHandler` docs](https://docs.rs/lightning/*/lightning/util/events/trait.EventHandler.html)
 
-### 17. Initialize the `ChannelManagerPersister`
+### 17. Initialize the `Persister`
 
 **What it's used for:** keeping `ChannelManager`'s stored state up-to-date
```
Comment (jkczyz) [inline docs/tutorials/build_a_node_in_rust.md:698]: Could you mention `NetworkGraph` here and under "Dependencies" below?

File: docs/tutorials/build_a_node_in_rust.md [hunk 2]
```diff
@@ -693,28 +693,37 @@ fn handle_ldk_event(..) {
...
 
 **Example:**
 
 ```rust
-// In this example, we provide a closure to handle ChannelManager persistence.
-// But you're also free to provide an implementation of the trait
-// `ChannelManagerPersister` instead.
+// In this example, we provide an implementation of the trait `Persister`
+
+struct YourDataPersister()
```
Comment (jkczyz) [inline docs/tutorials/build_a_node_in_rust.md:705]: Will need to make this struct definition match the one used in the `let persister = DataPersister { data_dir: ldk_data_dir.clone() };` line below.

File: docs/tutorials/build_a_node_in_rust.md [hunk 3]
```diff
@@ -693,28 +693,37 @@ fn handle_ldk_event(..) {
...
+
+impl Persister for YourDataPersister {
+	fn persist_manager(&self, channel_manager: &ChannelManager) -> Result<(), std::io::Error> {
+		// <insert code to persist the ChannelManager to disk and/or backups>
+	}
+
+	fn persist_graph(&self, network_graph: &NetworkGraph) -> Result<(), std::io::Error> {
+	   // <insert code to persist the NetworkGraph to disk and/or backups>
```
Comment (jkczyz) [inline docs/tutorials/build_a_node_in_rust.md:713]: nit: use a tab for the second indentation

**Comment by ConorOkus** [comment]
Sorry @jkczyz just catching up to this after the conference, how's it looking?

File: docs/tutorials/build_a_node_in_rust.md [hunk 4]
```diff
@@ -702,15 +702,17 @@ fn handle_ldk_event(..) {
 ```rust
 // In this example, we provide an implementation of the trait `Persister`
 
-struct YourDataPersister()
+struct YourDataPersister {
+	data_dir: String
+}
 
 impl Persister for YourDataPersister {
```
Comment (jkczyz) [inline docs/tutorials/build_a_node_in_rust.md:709]: Could you name this `DataPersister` to match the use on line 720?

File: docs/tutorials/build_a_node_in_rust.md [hunk 5]
```diff
@@ -721,7 +723,7 @@ let persister = DataPersister { data_dir: ldk_data_dir.clone() };
 
 **Implementation notes:** if the `ChannelManager` is not persisted properly to disk, there is risk of channels force closing the next time LDK starts up. However, in this situation, no funds other than those used to pay force-closed channel fees are at risk of being lost.
 
-**Dependencies:** `ChannelManager`
+**Dependencies:** `ChannelManager`, `NetworkGraph`
 
 **References:** [`Persister` docs](https://docs.rs/lightning-background-processor/*/lightning_background_processor/trait.Persister.html)
```
Comment (jkczyz) [inline docs/tutorials/build_a_node_in_rust.md:729]: On line 736 below, it looks like we're missing the `net_graph_msg_handler` parameter to `BackgroundProcessor::start`. Probably has been missing for awhile. 😬

Comment (ConorOkus) [reply docs/tutorials/build_a_node_in_rust.md:729]: Updated. My understanding is that it is optional though?

Comment (jkczyz) [reply docs/tutorials/build_a_node_in_rust.md:729]: Yes, optional but would need to pass `None` if not needed.

**Review by jkczyz — APPROVED** [review]

**Pull request merged**